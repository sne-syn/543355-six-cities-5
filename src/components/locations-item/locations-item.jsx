import PropTypes from 'prop-types';
import React from 'react';
import {AppRoute} from '../../utils/const';
import {Link} from 'react-router-dom';
import {changeActiveElement} from '../../store/action';
import {connect} from 'react-redux';
import {getActiveElement} from '../../store/offers-data/offers-data-selectors';

const LocationsItem = ({cityName, activeCity, tab, changeLocation}) => {
  let locationItemClass = `locations__item-link`;
  locationItemClass += (tab && cityName === activeCity) ? ` tabs__item--active tabs__item` : ``;
  locationItemClass += (tab) ? ` tabs__item` : ``;

  return (
    <Link to={AppRoute.ROOT} className={locationItemClass} onClick={changeLocation}>
      <span>{cityName}</span>
    </Link>
  );
};

LocationsItem.defaultProps = {
  activeCity: ``,
  tab: false
};

LocationsItem.propTypes = {
  cityName: PropTypes.string.isRequired,
  activeCity: PropTypes.string.isRequired,
  tab: PropTypes.bool.isRequired,
  changeLocation: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    activeCity: getActiveElement(state),
  };
}

const mapDispatchToProps = (dispatch) => ({
  changeLocation(evt) {
    dispatch(changeActiveElement(evt));
  }
});

export {LocationsItem};
export default React.memo(connect(mapStateToProps, mapDispatchToProps)(LocationsItem));
