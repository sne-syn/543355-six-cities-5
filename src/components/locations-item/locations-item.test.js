import {MemoryRouter} from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import {LocationsItem} from './locations-item';

const noop = () => {};

// cityName, activeCity, tab, changeLocation
describe(`<LocationsItem /> snapshots`, () => {
  it(`renders LocationsItem with default props`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <LocationsItem cityName={`Amsterdam`} changeLocation={noop} />
        </MemoryRouter>)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`renders LocationsItem with props`, () => {
    const tree = renderer
    .create(
        <MemoryRouter>
          <LocationsItem activeCity={`Paris`} cityName={`Amsterdam`} tab={true} changeLocation={noop} />
        </MemoryRouter>)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
