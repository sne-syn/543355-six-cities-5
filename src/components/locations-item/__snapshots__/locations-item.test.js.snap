// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationsItem /> snapshots renders LocationsItem with default props 1`] = `
<a
  className="locations__item-link"
  href="/"
  onClick={[Function]}
>
  <span>
    Amsterdam
  </span>
</a>
`;

exports[`<LocationsItem /> snapshots renders LocationsItem with props 1`] = `
<a
  className="locations__item-link tabs__item"
  href="/"
  onClick={[Function]}
>
  <span>
    Amsterdam
  </span>
</a>
`;
